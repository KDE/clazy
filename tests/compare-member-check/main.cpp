#include <tuple>

class ExampleClass
{
public:
    int a;
    int b;
    int c;

    bool operator==(const ExampleClass &other) const
    {
        return a == other.a || b == other.b || c == other.c; // No Warning, Using all the member variables in comparison
    }

    bool operator>(const ExampleClass &other) const
    {
        return a > other.a || b > other.b; // Warning, Using only two member variables in the comparison
    }

    bool operator<(const ExampleClass &rhs) const
    {
        return std::tie(a, b, c) < std::tie(rhs.a, rhs.b, rhs.c); // No Warning, Using all the member variables in comparison
    }
    bool operator>=(const ExampleClass &rhs) const
    {
        return std::tie(a) >= std::tie(rhs.a); // Warning, Using only one member variable in comparison
    }
};

int main()
{
    ExampleClass object1;
    ExampleClass object2;

    object1.a = 5;
    object1.b = 10;
    object1.c = 15;

    object2.a = 5;
    object2.b = 20;
    object2.c = 30;

    if (object1 == object2) { // Should not trigger a warning from the checker
    }

    if (object1 > object2) { // Should trigger a warning from the checker
    }
    if (object1 < object2) { // Should not trigger a warning from the checker
    }
    if (object1 >= object2) { // Should trigger a warning from the checker
    }

    return 0;
}