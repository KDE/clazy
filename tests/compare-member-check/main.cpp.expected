compare-member-check/main.cpp:46:17: warning: Comparison operator does not use all member variables [-Wclazy-compare-member-check]
compare-member-check/main.cpp:50:17: warning: Comparison operator does not use all member variables [-Wclazy-compare-member-check]
